chromadb/test/property/test_persist.py

```diff
@@ -239,3 +244,55 @@ def test_persist_embeddings_state(
     run_state_machine_as_test(
         lambda: PersistEmbeddingsStateMachine(settings=settings, client=client),
     )  # type: ignore
+
+
+# Ideally this scenario would be exercised by Hypothesis, but most runs don't seem to trigger this particular state.
```

Comment:
I'm not entirely sure why this is, and somewhat surprising to me as Hypothesis seems to bias towards smaller values. If I change `max_hnsw_batch_size`/`max_hnsw_sync_threshold` to both be 3, it catches this bug.